src/libraries/InterestRateModule.sol

```diff
@@ -0,0 +1,60 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity ^0.8.13;
+
+import {Owned} from "../../lib/solmate/src/auth/Owned.sol";
+import {DataTypes} from "./DataTypes.sol";
+
+abstract contract InterestRateModule is Owned {
+    uint256 public interestRate; //18 decimals precision
+
+    DataTypes.InterestRateConfiguration public interestRateConfig;
+
+    /**
+     * @notice Set's the configration parameters of InterestRateConfiguration struct
+     * @param newConfig New set of configration parameters
+     */
+
+    function setInterestConfig(DataTypes.InterestRateConfiguration calldata newConfig) external onlyOwner {
+        interestRateConfig = newConfig;
+    }
+
+    /**
+     * @notice Calculates the interest rate
+     * @param utilisation Utilisation rate
+     * @dev This function is only be called by the function _updateInterestRate(uint256 realisedDebt_, uint256 totalRealisedLiquidity_)
+     * @return Interest rate
+     */
+    function calculateInterestRate(uint256 utilisation) internal view returns (uint256) {
+        if (utilisation >= interestRateConfig.utilisationThreshold) {
+            uint256 lowSlopeInterest = interestRateConfig.utilisationThreshold * (interestRateConfig.lowSlope / 10 ** 5);
```

Comment:
Exponents are very gas intensive calculations, never use exponents for known parameters, use 100 000 instead of 10**5

Reply:
after checking in remix strangely lowest calculation for these was values with 'underscore' 

`pragma solidity ^0.8.7;

contract ZekiLab {
    // value input for each function was set to 100
    function calc1(uint256 value) public returns (uint256) { // gas: 25384, exec: 22073
        uint256 result = value * 10 ** 5;
        return result; // 10_000_000
    }

    function calc2(uint256 value) public returns (uint256) { // gas: 25359, exec: 22051
        uint256 result = value * 10e5;
        return result; // 10_000_000
    }

    function calc3(uint256 value) public returns (uint256) { // gas: 25309, exec: 22007
        uint256 result = value * 100_000;
        return result; // 10_000_000
    }

    function calc4(uint256 value) public returns (uint256) { // gas: 25334, exec: 22029
        uint256 result = value * 100000;
        return result; // 10_000_000
    }

}`

for future referance

Reply:
Done!